From 6139826011092a4794ddb9c73b2cecf6e53c2795 Mon Sep 17 00:00:00 2001
From: Kristoffer Flores <kflores@blackberry.com>
Date: Fri, 19 Jul 2013 13:57:49 -0400
Subject: [PATCH 16/61] Adde a manual test to verify the support of unicode
 characters when invoking cards

Reviewed by Rowell Cruz <rcruz@blackberry.com>
Tested by Tracy Li <tli@blackberry.com>
---
 test/integration/manual/blackberry.invoke.card.js |   75 +++++++++++++++++++++
 test/test-app/src/manual/framework/SpecRunner.htm |    2 +
 2 files changed, 77 insertions(+)
 create mode 100644 test/integration/manual/blackberry.invoke.card.js

diff --git a/test/integration/manual/blackberry.invoke.card.js b/test/integration/manual/blackberry.invoke.card.js
new file mode 100644
index 0000000..f4c2f6d
--- /dev/null
+++ b/test/integration/manual/blackberry.invoke.card.js
@@ -0,0 +1,75 @@
+/**
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+describe("email composer card", function () {
+    var content,
+        onSuccess,
+        onError,
+        onSuccessFlag,
+        onErrorFlag,
+        delay = 7500,
+        confirm;
+
+    beforeEach(function () {
+        onSuccessFlag = false;
+        onErrorFlag = false;
+        onSuccess = jasmine.createSpy("success callback").andCallFake(
+            function () {
+                onSuccessFlag = true;
+            }
+        );
+        onError = jasmine.createSpy("fail callback").andCallFake(
+            function () {
+                onErrorFlag = true;
+            }
+        );
+    });
+
+    afterEach(function () {
+        onSuccess = null;
+        onError = null;
+        onSuccessFlag = false;
+        onErrorFlag = false;
+    });
+
+    it("should show unicode characters", function () {
+        content = {
+            to : ["ňƎƊstark@winterfell.ca"],
+            cc : ["ȐȎƁstark@winterfell.com"],
+            subject : "A Stark µ",
+            body : "ƙing of the Ņorth",
+            attachment : ["/some/path.txt", "local:///other/path.txt"]
+        };
+
+        try {
+            blackberry.invoke.card.invokeEmailComposer(content, onSuccess, onError);
+        } catch (e) {
+            console.log(e);
+        }
+
+        waitsFor(function () {
+            return onSuccessFlag || onErrorFlag;
+        }, "The callback flag should be set to true", delay);
+
+        runs(function () {
+            confirm = window.confirm("Did unicode characters appear in the fields?");
+            expect(confirm).toEqual(true);
+        });
+    });
+});
diff --git a/test/test-app/src/manual/framework/SpecRunner.htm b/test/test-app/src/manual/framework/SpecRunner.htm
index 0211b23..7881eb7 100644
--- a/test/test-app/src/manual/framework/SpecRunner.htm
+++ b/test/test-app/src/manual/framework/SpecRunner.htm
@@ -2,6 +2,7 @@
 <html>
 <head>
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
     <title>Jasmine Spec Runner</title>
 
     <link rel="stylesheet" type="text/css" href="../../css/jasmine.css">
@@ -83,6 +84,7 @@
         <!--include spec files here-->
         <input type="checkbox" value="blackberry.invoke">blackberry.invoke</input><br />
         <input type="checkbox" value="blackberry.invoked">blackberry.invoked</input><br />
+        <input type="checkbox" value="blackberry.invoke.card">blackberry.invoke.card</input><br />
         <input type="checkbox" value="blackberry.ui.dialog">ui.dialog</input><br />
         <input type="checkbox" value="blackberry.io.filetransfer">blackberry.io.filetransfer</input><br />
         <input type="checkbox" value="blackberry.push">blackberry.push</input><br />
-- 
1.7.10.msysgit.1

